src/card/color.rs

```diff
@@ -184,12 +182,45 @@ impl Colors {
     /// Checks if this instance is multicolored, which is true if it contains
     /// more than one color flag.
     pub const fn is_multicolored(self) -> bool {
-        self.0.count_ones() >= 2
+        self.0.count_ones() > 1
     }
 
     /// Checks if this instance is colorless.
     pub const fn is_colorless(self) -> bool {
-        self.0 == 0
+        self.0 == Colorless as u8
+    }
+
+    /// Produces a new instance with all the colors from both `self` and
+    /// `other`.
+    pub const fn union(self, other: Colors) -> Self {
+        Colors(self.0 | other.0)
+    }
+
+    /// Produces a new instance with the colors that `self` and `other` have in
+    /// common.
+    pub const fn intersection(self, other: Colors) -> Self {
+        Colors(self.0 & other.0)
+    }
+
+    /// Produces a new instance with the colors from `other` removed.
+    pub const fn difference(self, other: Colors) -> Self {
+        Colors(self.0 & !other.0)
+    }
+
+    /// Produces a new instance with the colors that are in `self` or `other`,
+    /// but not both.
+    pub const fn symmetric_difference(self, other: Colors) -> Self {
+        Colors((self.0 ^ other.0) & (self.0 | other.0))
+    }
+
+    /// Produces a new instance with the specified color added.
+    pub const fn with(self, color: Color) -> Self {
+        Colors(self.0 | color as u8)
+    }
+
+    /// Produces a new instance with the specified color removed.
+    pub const fn without(self, color: Color) -> Self {
+        Colors(self.0 & !(color as u8))
```

Comment:
Isn't this wrong?
won't this set the highest 3 bits?

Reply:
No, I don't think so. `!(color as u8)` will have its highest 3 bits set, but `self.0` never will, so since the operation is `&` this should be correct.